pipeline.py

```diff
@@ -72,26 +85,60 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+        tokens = []
+        for article in articles:
+            article_raw_text = article.get_raw_text()
+
+            for token in self._process(article_raw_text):
+                tokens.append(token.get_cleaned())
+
+            article.save_as(' '.join(tokens), ArtifactType.cleaned)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+
+        text = re.sub(r"[^а-я\s]", "", raw_text)
+        tokens = text.split()
+
+        morphological_tokens = [MorphologicalToken(original_word=word) for word in tokens]
+
+        return morphological_tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    path = Path(path_to_validate)
+
+    if not path.exists():
+        raise FileNotFoundError
+
+    if not path.is_dir():
+        raise NotADirectoryError
+
+    if not list(path.iterdir()):
+        raise EmptyDirectoryError
+
+    meta = []
+    raw = []
+    for file in path.rglob('*.json'):
+        meta.append(file.parts[-1].split('_')[0])
+    for file in path.rglob('*_raw.txt'):
+        raw.append(file.parts[-1].split('_')[0])
```

Comment:
the same about checking and extracting digits.

Reply:
changed